props/saber_fett263_buttons.h

```diff
@@ -293,248 +664,3398 @@ class SaberFett263Buttons : public virtual PropBase {
+        MenuRevert();
+        break;
+#if NUM_BLADES > 1
+      case MENU_BLADE_STYLE:
+        menu_type_ = MENU_TOP;
+        show_preview_.Stop(blade_preview_);
+        blade_preview_ = 0;
+        MenuCancel();
+        break;
+      case MENU_BLADE_COLOR:
+        menu_type_ = MENU_TOP;
+        show_preview_.Stop(blade_preview_);
+        blade_preview_ = 0;
+        MenuCancel();
+        break;
+      case MENU_BLADE_COPY:
+        menu_type_ = MENU_COLOR_SUB;
+        show_preview_.Stop(blade_preview_);
+        blade_preview_ = 0;
+        MenuCancel();
+        break;
+      case MENU_BLADE_LENGTH:
+        menu_type_ = MENU_SETTING_SUB;
+        show_preview_.Stop(blade_preview_);
+        blade_preview_ = 0;
+        MenuCancel();
+        break;
+      case MENU_COPY_COLOR:
+        menu_type_ = MENU_COLOR_SUB;
+        show_preview_.Stop(blade_preview_);
+        blade_preview_ = 0;
+        MenuCancel();
+        break;
+#endif
+      case MENU_STYLE:
+        char style_arg[10];
+        itoa(style_revert_, style_arg, 10);
+        current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), 1, style_arg)); 
+        current_preset_.Save();
+        UpdateStyle(current_preset_.preset_num);
+#if NUM_BLADES == 1
+        menu_type_ = MENU_TOP;
+        MenuRevert();
+#else
+        menu_type_ = MENU_BLADE_STYLE;
+        MenuRevert();
+#endif
+        break;
+      case MENU_COLOR:
+        menu_type_ = MENU_TOP;
+        SaberBase::SetVariation(variation_revert_);
+        ToggleColorChangeMode();
+        MenuCancel();
+        break;
+      case MENU_COLOR_SUB:
+#if NUM_BLADES == 1
+        menu_type_ = MENU_TOP;
+        MenuCancel();
+        break;
+#else
+        menu_type_ = MENU_BLADE_COLOR;
+        show_preview_.Stop(blade_preview_);
+        blade_preview_ = 0;
+        blade_num_ = 0;
+        MenuCancel();
+        break;
+#endif
+      case MENU_COLOR_MODE:
+        edit_color_ = false;
+        menu_type_ = MENU_EFFECT;
+        MenuCancel();
+        break;
+      case MENU_EFFECT:
+        edit_color_ = false;
+        menu_type_ = MENU_COLOR_SUB;
+        MenuCancel();
+        break;
+      case COLOR_BASE:
+      case COLOR_ALT:
+      case COLOR_IGNITE:
+      case COLOR_RETRACT:
+      case COLOR_SWING:
+      case COLOR_OFF:
+        menu_type_ = MENU_COLOR_MODE;
+        edit_color_ = false;
+        twist_menu_ = M_PI / 4;
+        show_color_.Stop(blade_num_);
+        MenuRevert();
+        color_mode_ = NONE;
+        break;
+      case COLOR_BLAST:
+      case COLOR_CLASH:
+      case COLOR_LOCKUP:
+      case COLOR_LB:
+        menu_type_ = MENU_COLOR_MODE;
+        color_mode_ = NONE;
+        edit_color_ = false;
+        twist_menu_ = M_PI / 4;
+        bump_color_.Stop(blade_num_);
+        MenuRevert();
+        color_mode_ = NONE;
+        break;
+      case COLOR_DRAG:
+      case COLOR_STAB:
+        menu_type_ = MENU_COLOR_MODE;
```

Comment:
All of these look the same, except for which thing they call Stop() on.
I suggest taking out all the common code and put it in a helper function, like:

void MakeABetterNameForThisFunction() {
  color_mode_ = NONE;
  edit_color_ = false;
  twist_menu_ = M_PI / 4;
  color_mode_ = NONE;
}

Then these cases all become something like:

    tip_color_.Stop(blade_num_);
    MakeABetterNameForThisFunction();

Reply:
added SaveColorEdit() and RevertColorEdit() helper functions